storage/range.go

```diff
@@ -509,12 +592,32 @@ func (r *Range) addReadWriteCmd(method string, args proto.Request, reply proto.R
 		log.Errorf("unable to read result for %+v from the response cache: %s", args, err)
 	}
 
+	// For read/write commands, we check the leader lease. If we know of
+	// a lease and the holder is not this replica, and we're within the
+	// lease expiration, return NotLeaderError. In all other cases, we
+	// want to continue. Either we're the lease holder, or were the last
+	// lease holder, or we weren't the last but we'd like to become the
+	// leader because this replica is where there's write pressure.
```

Comment:
It's important to maintain the distinction between "leader" and "lease holder". Raft will continually try to ensure that there is a leader (unless and until we implement quiescent ranges #357), but we are allowing the lease to go dormant. Consistent reads should be forwarded or redirected to the leader regardless of the lease status, and then the leader may have to reacquire the lease.